### ISSUE TITLE
[BUG] Action-listops deletes all tags when there is an exclamation mark at end of title.

### ISSUE THREAD

**Issue by Marxsal** [Jermolene/TiddlyWiki5#5576]
To recreate, create a tiddler without an exclamation mark with two tags: "AnyTag", "Tag1"

Insert button code:

```
<$button>Remove
<$action-listops  $tags="-[[Tag1]]"/>
</$button>

```

This works fine, removing just Tag1. Now repeat, but this time put an exclamation mark at the end of the tiddler title (or actually, possibly anywhere in the title).

Now when you run it, it will remove ALL tags from the current tiddler.

Also, a Double exclamation mark anywhere in the title will cause the same problem.

See thread at 
https://www.reddit.com/r/TiddlyWiki5/comments/may2bu/bug_or_user_error_actionlistops_stripping_all/

**Comment by saqimtiaz** [comment]
This seems to be where the problem lies:
https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/widgets/action-listops.js#L85

`tagfilter = "[list[" + this.target + "!!tags]] " + this.filtertags,`

can be changed to:

`tagfilter= "[[" + this.target + "]tags[]] " + this.filtertags`

**Comment by Jermolene** [comment]
> This seems to be where the problem lies:
> https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/widgets/action-listops.js#L85
> 
> `tagfilter = "[list[" + this.target + "!!tags]] " + this.filtertags,`
> 
> can be changed to:
> 
> `tagfilter= "[[" + this.target + "]tags[]] " + this.filtertags`

Aaaargh! The action-listops widget has always been a disaster, I've long regretted merging it.

We need to rewrite it to not rely on stitching filter expressions together. It looks feasible but it's going to be fiddly.

**Comment by pmario** [comment]
@Marxsal ... IMO you can close this issue?

**Comment by Marxsal** [comment]
Fixed in #5644
